src/main/java/com/demoipm/controller/UserManageController.java

```diff
@@ -0,0 +1,150 @@
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+
+import javax.validation.Valid;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Controller
+public class UserManageController {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(UserManageController.class);
+
+    @Autowired
+    private UserAppService userAppService;
+
+    @Autowired
+    private RoleAppService roleAppService;
+
+    @Secured("ROLE_ADMIN")
+    @RequestMapping("/manage-user")
+    public String getUserList(@RequestParam(value = "searchWord", required = false) String searchWord,
+                              @RequestParam(value = "pageNo", required = false) Integer pageNo,
+                              @RequestParam(value = "entriesNo", required = false) Integer entriesNo,
+                              Model model) {
+        LOGGER.info("Start get user list with search word {}, page {}, entries no {}", searchWord, pageNo, entriesNo);
+        entriesNo = entriesNo == null || entriesNo <= 0 ? 10 : entriesNo;
+        pageNo = pageNo == null || pageNo <= 0 ? 1 : pageNo;
+        searchWord = searchWord != null ? searchWord : "";
+        UserListPageResponseDto responseDto = userAppService.readByCondition(searchWord, pageNo, entriesNo);
+        model.addAttribute("response", responseDto);
+        LOGGER.info("End get user list with search word {}, page {}, entries no {}", searchWord, pageNo, entriesNo);
+        return "manageuser/manage-user";
+    }
+
+    @Secured("ROLE_ADMIN")
+    @RequestMapping("/create-user-page")
+    public String createUserPage(Model model) {
+        LOGGER.info("Start show create user page");
+        UserCreateRequestDto user = new UserCreateRequestDto();
+        model.addAttribute("user", user);
+        initRolesForUserModifyPage(model);
+        return "manageuser/create-user";
+    }
+
+    @Secured("ROLE_ADMIN")
+    @PostMapping("/process-create-user")
+    public String processCreateUser(@Valid @ModelAttribute("user") UserCreateRequestDto userCreateRequestDto,
+                                    BindingResult result,
+                                    Model model) {
+        LOGGER.info("Start process create user {}", userCreateRequestDto);
+        if (result.hasErrors()) {
+            initRolesForUserModifyPage(model);
+            return "manageuser/create-user";
+        }
+
+        ResponseDto responseDto = new ResponseDto();
+        userAppService.createNewUser(userCreateRequestDto, responseDto);
+        if (responseDto.hasError()) {
+            initRolesForUserModifyPage(model);
+            model.addAttribute("response", responseDto);
+            return "manageuser/create-user";
+        }
+        return "redirect:manage-user";
+    }
+
+    private void initRolesForUserModifyPage(Model model) {
```

Comment:
Refactor this function to match with SOLID pattern.
Move this function to the end of code.

Reply:
Fixed